### ISSUE TITLE
[BUG] Uncaught warning: WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: mlflowdbfs:///artifact?run_id=foo&path=/bar.

### ISSUE THREAD

**Issue by philmassie** [mlflow/mlflow#6608]
### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

1.28.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Amazon Linux 2
- **Python version**: 3.7.11
- **yarn version, if running the dev UI**: Unknown
- Spark 3.2.1

### Describe the problem

Hello
I dont know if this belongs here or in Spark, but I'll start here.
The warning doesn't show up in earlier Spark version (I had 3.0.3 handy), but in 3.2.1
`mlflow.spark.log_model(model_trained, "spark-model")`
produces:
```
22/08/26 15:12:26 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: mlflowdbfs:///artifact?run_id=foo&path=/bar.
org.apache.hadoop.fs.UnsupportedFileSystemException: No FileSystem for scheme "mlflowdbfs"
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3342)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3362)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:123)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3413)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3381)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:486)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
	at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:53)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:370)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:274)
	at org.apache.spark.sql.DataFrameReader.$anonfun$load$3(DataFrameReader.scala:245)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:245)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:188)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
	at java.lang.Thread.run(Thread.java:750)
```

The model does get saved to S3, and parameters are logged successfully. but the long warnings ruin the iterative model training output.

Apologies if this should be in Spark, or if its not actually a bug.
If someone has a workaround please let me know, I'm trying to sell my team on MLFlow at the moment.

### Code to reproduce issue

Poking around the warning originates with the following function call
`_get_active_spark_session().read.load("mlflowdbfs:///artifact?run_id=foo&path=/bar")` within `mlflow.spark._should_use_mlflowdbfs(root_uri)` and can be reproduced on my system in a clean Conda environment with only  PySpark 3.2.1 installed:
```
try:
    spark.read.load("mlflowdbfs:///artifact?run_id=foo&path=/bar")
except Exception as e:
    print("Exception")
```
PySpark 3.0.3 doesn't throw the warning. I don't understand why the exception isn't getting caught by the Python try/catch.

### Stack trace

22/08/26 15:12:26 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: mlflowdbfs:///artifact?run_id=foo&path=/bar.
org.apache.hadoop.fs.UnsupportedFileSystemException: No FileSystem for scheme "mlflowdbfs"
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:3342)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3362)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:123)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:3413)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:3381)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:486)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:365)
	at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:53)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:370)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:274)
	at org.apache.spark.sql.DataFrameReader.$anonfun$load$3(DataFrameReader.scala:245)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:245)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:188)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
	at java.lang.Thread.run(Thread.java:750)

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by philmassie** [comment]
For the time being my workaround is to comment out the contents of `_should_use_mlflowdbfs()`, I'm sure that's not what you're supposed to do though.

**Comment by dbczumar** [comment]
Hi @philmassie, apologies for introducing this nuisance of a warning message. Fortunately, this has been addressed via https://github.com/mlflow/mlflow/pull/6474, which will be included in the MLflow 1.29.0 release. For now, if you install MLflow from master via `pip install git+https://github.com/mlflow/mlflow@master`, the warning should disappear. Thank you for using MLflow!
